src/Language/Futhark/TypeChecker/Terms.hs

```diff
@@ -269,6 +269,7 @@ checkApplyExp e = do
 checkExp :: UncheckedExp -> TermTypeM Exp
 checkExp (Literal val loc) =
   pure $ Literal val loc
+checkExp (Hole name t loc) = typeError loc mempty $ "Found typed hole: " <+> text (show name) <+> text (show t) <> "."
```

Comment:
Linebreak.  Also, don't bother printing the `t`, since it has no information at this point.